app/forms/supplier_user_form.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+class SupplierUserForm
+  include ActiveModel::Model
+  include ActiveModel::Serialization
+
+  attr_accessor :full_name, :email, :supplier
+  validates :supplier, presence: { message: "Select one" }, on: :supplier
+  validates :full_name, presence: { message: "Enter a name" }, on: :details
+  validates :email, presence: { message: "Enter email" }, on: :details
+  validate :email_not_taken, on: :details
+
+  def attributes
+    { "full_name" => nil, "email" => nil, "supplier" => nil }
+  end
+
+  def chosen_supplier
+    LeadProvider.find_by(id: supplier) || DeliveryPartner.find_by(id: supplier)
+  end
+
+  def save!
+    profile = if chosen_supplier.instance_of?(LeadProvider)
+                LeadProviderProfile.new(lead_provider: chosen_supplier)
+              else
+                DeliveryPartnerProfile.new(delivery_partner: chosen_supplier)
+              end
+
+    ActiveRecord::Base.transaction do
+      @user = User.create!(
+        full_name: full_name,
+        email: email,
+        confirmed_at: Time.zone.now.utc, # Skip confirmation email
```

Comment:
Why do we need to confirm their accounts? I'd have expected them to do this themselves, if anything to confirm the email address they will use to sign in.

Reply:
This is what was designed. The expectation is that this account may be created months before it's first used, so any confirmation link would expire. We also expect the email entered by the administrator to be correct

Reply:
See https://miro.com/app/board/o9J_ldVNkCY=/?openComment=3074457353853155754